pandas/core/series.py

```diff
@@ -4059,9 +4059,10 @@ def _try_cast(arr, take_fast_path):
     if issubclass(subarr.dtype.type, compat.string_types):
         # GH 16605
         # If not empty convert the data to dtype
-        if not isna(data).all():
-            data = np.array(data, dtype=dtype, copy=False)
-
-        subarr = np.array(data, dtype=object, copy=copy)
+        # GH 19853: If data is a scalar, subarr has already the result
+        if not is_scalar(data):
```

Comment:
is this additional check actually needed? (is_scalar)

Reply:
Yes. If it is scalar, no other change is needed, since `subarr` has the correct result. If it is not scalar, then it does the conversion 
```python
data = np.array(data, dtype=dtype, copy=False)
```
(That conversion was there already)

Reply:
my point is that do you actually need to add this check? when you take it out (but fix the ``np.any``) is there any problem?

Reply:
If I take out, then this https://github.com/pandas-dev/pandas/pull/20401/files#diff-3bbe4551f20de6060dce38a95d0adc80R114 will raise
```
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../core/series.py:270: in __init__
    data = SingleBlockManager(data, index, fastpath=True)
../core/internals.py:4632: in __init__
    block = make_block(block, placement=slice(0, len(axis)), ndim=1)
../core/internals.py:3161: in make_block
    return klass(values, ndim=ndim, placement=placement)
../core/internals.py:2268: in __init__
    placement=placement)
../core/internals.py:117: in __init__
    self.ndim = self._check_ndim(values, ndim)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <[AttributeError("ndim") raised in repr()] ObjectBlock object at 0x7f1ae266da98>
values = array('', dtype=object), ndim = 1

    def _check_ndim(self, values, ndim):
        """ndim inference and validation.
    
            Infers ndim from 'values' if not provided to __init__.
            Validates that values.ndim and ndim are consistent if and only if
            the class variable '_validate_ndim' is True.
    
            Parameters
            ----------
            values : array-like
            ndim : int or None
    
            Returns
            -------
            ndim : int
    
            Raises
            ------
            ValueError : the number of dimensions do not match
            """
        if ndim is None:
            ndim = values.ndim
    
        if self._validate_ndim and values.ndim != ndim:
            msg = ("Wrong number of dimensions. values.ndim != ndim "
                   "[{} != {}]")
>           raise ValueError(msg.format(values.ndim, ndim))
E           ValueError: Wrong number of dimensions. values.ndim != ndim [0 != 1]

../core/internals.py:153: ValueError
```